### ISSUE TITLE
[oracle] 'Found unparsable section' with rem[ark]

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8064]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wanted to run the linter on a file with the following example. This leads to the described error.

### Expected Behaviour

I would expect the example to be correctly parsed. The keyword default should be supported as well.
See https://docs.oracle.com/en/database/oracle/oracle-database/18/sqpug/REMARK.html

### Observed Behaviour

`Line 1, Position 1: Found unparsable section: 'rem abc'`

### How to reproduce

```sql
rem abc
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
